docs: `podman`: known issue + macOS doc (#698)
knative-sandbox/kn-plugin-func: docs/guides/podman.md

❯ func build -v
```
Now you may use `func` as usual.

## macOS

Under macOS you need to use `podman machine` as `podman` is native to Linux.

```
❯ brew install podman
❯ podman machine init --cpus=2 --disk-size=30 --memory=8192
❯ podman machine start
❯ ssh-add -k $HOME/.ssh/podman-machine-default
❯ export DOCKER_HOST=(podman system connection ls --format="{{.URI}}" | grep root)
```

## Known issues

* In func version 0.19, some runtimes might be unable to build a function by using podman. You might see an error message similar to the following:

    ```

>     ERROR: failed to image: error during connect: Get "http://%2Fvar%2Frun%2Fdocker.sock/v1.40/info": EOF

Reviewer: Does it really say `failed to image`?

Author: I copied that from osh doc.

Reviewer: lol - that's probably a typo, but who knows!?

Reviewer 2: yea.. it appears like that.. failed to image... lol

    ```
  * The following workaround exists for this issue:

>     * Update the podman service by adding --time=0 to the service ExecStart definition.  The configuration files are: `/etc/systemd/[user|system]/podman.service`.You might need to copy default vendor configuration files from `/usr/lib/systemd/[user|system]/podman.service` first.

Reviewer: ```suggestion
    * Update the podman service by adding --time=0 to the service ExecStart definition in the podman configuration file at `/etc/systemd/user/podman.service`.You might need to copy the default configuration file from `/usr/lib/systemd/user/podman.service` first.
```
Some minor wording tweaks. I wonder if it wouldn't be better to just refer to the  `user` config files, since that's probably the only thing they will need to modify.
